frontend/workflows/ec2/src/terminate-instance.tsx

```diff
@@ -109,7 +121,11 @@ const TerminateInstance: React.FC<WorkflowProps> = ({ heading, resolverType, not
 
   return (
     <Wizard dataLayout={dataLayout} heading={heading}>
-      <InstanceIdentifier name="Lookup" resolverType={resolverType} />
+      <InstanceIdentifier
+        name="Lookup"
+        resolverType={resolverType}
+        inputTransformer={inputTransformer}
```

Comment:
it would be nice if the workflow automatically plugged this into a context somehow so we wouldn't have to prop drill every workflow. is that possible?

Reply:
hm not sure, I'll think about this and maybe ask derek as well

Reply:
Synced with Derek, came to a couple of conclusions:

There are 2 separate issues here - 1. the publishing of the config, and 2. the consumption of the config. We will focus on the latter.
- For prop drilling, we know the types with our well defined interface, whereas if we were pulling from a context we would need to store the type information somewhere else. For example, we know what `notes` is at all times in each workflow. We could move this interface to a higher level (i.e the context)
- Potential Issues of re-rendering - how do we position the context properly to prevent re-renders down the component tree? This would seem to point to a per-workflow rather than global context.
- For a context approach, we could infer values based off a workflow name. Is that an assumption we want to go with?

In general, it is certainly doable to have a global or per workflow context from which we pull objects using hooks. It is a question that becomes one of prioritization as well as design.

Reply:
I tried searching around for this topic and I found "component composition" in this: https://blog.logrocket.com/solving-prop-drilling-react-apps/ but I'm not sure that's a good solution either.

Reply:
> Potential Issues of re-rendering - how do we position the context properly to prevent re-renders down the component tree? This would seem to point to a per-workflow rather than global context.

Wouldn't the context be static and read-only after initialization of the application?

Reply:
we synced, but when I tried Daniel's solution from the sync I realized that we can't read it in from the config anymore.
Doc explaining the process https://docs.google.com/document/d/1kmLHmhCls6GbMvju-3CI9ajhMQqMTj53H_1EQB4R8QA/edit?usp=sharing